cognite/client/_api/assets.py

```diff
@@ -1006,6 +1010,8 @@ def list(
         aggregated_properties: Sequence[AggregateAssetProperty] | None = None,
         partitions: int | None = None,
         limit: int | None = DEFAULT_LIMIT_READ,
+        advanced_filter: Filter | dict | None = None,
```

Comment:
If we are gonna do this, it should be done across the board - for all resource types supporting advanced filters. I'd like your input on whether we should do this at all though @haakonvt @doctrino.